src/functions_framework/event_conversion.py

```diff
@@ -147,13 +152,30 @@ def background_event_to_cloudevent(request) -> CloudEvent:
             uid = data["uid"]
             subject = f"users/{uid}"
 
+    # Handle Firebase DB events.
+    if service == _FIREBASE_DB_CE_SERVICE:
+        # The CE source of firebasedatabase cloudevents includes location information
+        # that is inferred from the 'domain' field of legacy events.
+        if "domain" not in event_data:
+            raise EventConversionException(
+                "Invalid FirebaseDB event payload: missing 'domain'"
+            )
+
+        domain = event_data["domain"]
+        location = "us-central1"
+        if domain != "firebaseio.com":
+            location = domain.split(".")[0]
+
+        resource = f"projects/_/locations/{location}/{resource[11:]}"
```

Comment:
This seems fragile, what does `11` represent here? Would it make more sense to update the `_split_resource` function or the `_CE_SERVICE_TO_RESOURCE_RE` regex?

Reply:
This is a good point. I will update the regex.